parity-crypto/src/publickey/secret_key.rs

```diff
@@ -232,7 +238,7 @@ impl TryFrom<&[u8]> for Secret {
 		if b.len() != SECP256K1_SECRET_KEY_SIZE {
```

Comment:
Both `TryFrom` implementations can't really guarantee that the source slice is zeroized, there is no way to inform the user that it should be.
Not a security expert, but perhaps it's worth to get rid of these in favour of some more explicit methods which would allow you to add extra security notes.

Reply:
I am more having an issue with the From implementation, `try_from` never zeroized the input, but yes we do not have the small doc.
I will be in favor of replacing those from or try_from implementation with specific function, but then it would be a breaking PR, so I am thinking this PR could be parity-crypto 6.2 and another one could be a 7.0 with a more significant work for library users.

Reply:
Actually that is not such a big change, maybe doing it now and releasing breaking 0.7.0 would be better, I will do it.

Reply:
Sounds good to me!

Reply:
I am a big fan of deprecation warnings, so people get a hint on how to migrate directly from the build - just an idea though

Reply:
Could try 6.2 with dep warning indeed.